jenkins-slave

```diff
@@ -75,8 +75,20 @@ else
 		fi
 	fi
 
+	# JENKINS_CA_CERT points a path for ca cert (possibly mounted as a secret volume)
+	if [ -n "$JENKINS_CA_CERT" ]; then
+		jre_keystore=/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts
+		keystore_dir=$HOME/.keystore
+		mkdir -p $keystore_dir
+		cp $jre_keystore $keystore_dir
+		cat $JENKINS_CA_CERT > /tmp/cacert
+		keytool -import -keystore $keystore_dir/cacerts -alias master -file /tmp/cacert -storepass changeit -noprompt
+		rm /tmp/cacert
+		TRUSTSTORE="-Djavax.net.ssl.trustStore=$HOME/.keystore/cacerts -Djavax.net.ssl.trustStorePassword=changeit"
+	fi
+
 	#TODO: Handle the case when the command-line and Environment variable contain different values.
 	#It is fine it blows up for now since it should lead to an error anyway.
 
-	exec java $JAVA_OPTS $JNLP_PROTOCOL_OPTS -cp /usr/share/jenkins/slave.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL $OPT_JENKINS_SECRET $OPT_JENKINS_AGENT_NAME "$@"
+	exec java $JAVA_OPTS $TRUSTSTORE $JNLP_PROTOCOL_OPTS -cp /usr/share/jenkins/slave.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL $OPT_JENKINS_SECRET $OPT_JENKINS_AGENT_NAME "$@"
```

Comment:
I think that can work. I am using this with the Kubernetes plugin. Using this approach will mean that volume need to be prepared in Java's trust store format instead of PEM file. The benefit of PEM format is that, I can use the same secret is other places etc.

My question is what is stopping user job changing /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts as it stands today (without the JENKINS_CA_CERT env)?

Also, java process is PID 1 inside the container. How do you see user process messing with that?